### PR TITLE
Information screen

### PR DESCRIPTION
**Pull request by sam33rdhakal**
Updates
* Adds missing UI elements from the information screen
* Adds join now feature (launches custom tab)

### CODE REVIEW

File: base/src/main/java/edu/artic/base/utils/LifecycleAware.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    fun onStart(host: T)
+
+    @OnLifecycleEvent(ON_STOP)
+    fun onStop(host: T)
+
+    @OnLifecycleEvent(ON_DESTROY)
+    fun onDestroy(host: T)
+}
+
+interface FragmentLifecycleAware : LifecycleAware<Fragment>
```
Comment (raigex) [inline base/src/main/java/edu/artic/base/utils/LifecycleAware.kt]: Please rename to LifecycleAwareFragment more gramatically correct and less likely to stop me from reading

File: base/src/main/java/edu/artic/base/utils/LifecycleAware.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+interface ActivityLifecycleAware : LifecycleAware<Activity>
```
Comment (raigex) [inline base/src/main/java/edu/artic/base/utils/LifecycleAware.kt]: same as above

File: info/src/main/java/edu/artic/info/InformationFragment.kt [hunk 3]
```diff
@@ -27,23 +38,33 @@ class InformationFragment : BaseViewModelFragment<InformationViewModel>() {
...
                                 }
+                                is InformationViewModel.NavigationEndpoint.JoinNow -> {
+                                    val url = (it.endpoint as InformationViewModel.NavigationEndpoint.JoinNow).url
+                                    customTabManager.openUrlOnChromeCustomTab(requireContext(), Uri.parse(resources.getString(url)))
+                                }
                             }
                         }
                     }
-                }
+                }.disposedBy(disposeBag)
```
Comment (raigex) [inline info/src/main/java/edu/artic/info/InformationFragment.kt]: change to use navigationdisposebag

File: info/src/main/java/edu/artic/info/InformationViewModel.kt [hunk 4]
```diff
@@ -11,10 +12,15 @@ class InformationViewModel @Inject constructor() : NavViewViewModel<InformationV
 
     sealed class NavigationEndpoint {
         object AccessMemberCard : NavigationEndpoint()
-        object Search: NavigationEndpoint()
+        object Search : NavigationEndpoint()
+        class JoinNow(@StringRes val url: Int) : NavigationEndpoint()
```
Comment (raigex) [inline info/src/main/java/edu/artic/info/InformationViewModel.kt]: Pull join url from dataObjectDao

File: info/src/main/java/edu/artic/info/InformationViewModel.kt [hunk 5]
```diff
@@ -11,10 +12,15 @@ class InformationViewModel @Inject constructor() : NavViewViewModel<InformationV
...
     }
 
     fun onClickSearch() {
         navigateTo.onNext(Navigate.Forward(NavigationEndpoint.Search))
     }
+
+    fun joinNow() {
+        navigateTo.onNext(Navigate.Forward(NavigationEndpoint.JoinNow(R.string.joinUrl)))
```
Comment (raigex) [inline info/src/main/java/edu/artic/info/InformationViewModel.kt]: useDataObjectDao for this url not strings file

File: info/src/main/java/edu/artic/info/InformationFragment.kt (hunk 3 above)
Comment (sam33rdhakal) [reply info/src/main/java/edu/artic/info/InformationFragment.kt]: Updated.

File: base/src/main/java/edu/artic/base/utils/LifecycleAware.kt (hunk 2 above)
Comment (sam33rdhakal) [reply base/src/main/java/edu/artic/base/utils/LifecycleAware.kt]: Updated.

File: base/src/main/java/edu/artic/base/utils/LifecycleAware.kt (hunk 1 above)
Comment (sam33rdhakal) [reply base/src/main/java/edu/artic/base/utils/LifecycleAware.kt]: Done.

**Pull request merged**